[Inductor] Include math and torch in launcher scope

muchulee8 opened the pull request:
Summary:
For grid computation, if we have sympy, it is possible we have math and torch used.
We include the math and torch module in the launcher scope to make sure those grid get computed correctly.

Test Plan: Check phabricator for internal cmd.

Differential Revision: D75642931




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @amjames @chauhang @aakhundov

eellison reviewed (commented):
Can you give more context, or include a test ? when would we use torch

muchulee8 committed:
[Inductor] Include math and torch in launcher scope (#154673)

Summary:

For grid computation, if we have sympy, it is possible we have math and torch used.
We include the math and torch module in the launcher scope to make sure those grid get computed correctly.

Test Plan:
```
buck2 run mode/opt  mode/inplace -c fbcode.platform010_cuda_version=12 -c fbcode.nvcc_arch=h100 caffe2/torch/fb/model_transform/experimental/benchmark:mts_gpu_benchmark -- --lower-backend=AOT_INDUCTOR   --model-path=manifold://ads_storage_fblearner/tree/user/facebook/fblearner/predictor/739097956/0/gpu_lowering/input.predictor.disagg.gpu.merge --trace-aot-inductor-module=True --disable-acc-tracer=False --batch-size=2048 --node_replacement_dict "{'torch.nn.Linear':{'(4000+,1000+)':'fp8_float_model_dynamic_quantization_rowwise','(0+,2000+)':'fp8_float_model_dynamic_quantization_rowwise'}}" --use_dynamic_dim_hints=True --allow_refine_dynamic_shapes_on_constants=True
```

Differential Revision: D75642931

davidberard98 reviewed (approved):
thanks for the fix & the test!

muchulee8 commented:
@pytorchbot merge -f "Merged in Phabricator already."
